src/js/components/MaskedInput/doc.js

```diff
@@ -0,0 +1,46 @@
+import { describe, PropTypes } from 'react-desc';
+
+import { getAvailableAtBadge } from '../../utils';
+
+export const doc = MaskedInput => {
+  const DocumentedMaskedInput = describe(MaskedInput)
+    .availableAt(getAvailableAtBadge('MaskedInput'))
+    .description('An input field with formalized syntax.')
+    .usage(
+      `import { MaskedInput } from 'grommet';
+<MaskedInput id='item' name='item' />`,
+    );
+
+  DocumentedMaskedInput.propTypes = {
+    id: PropTypes.string.description('The id attribute of the input.'),
+    name: PropTypes.string.description('The name attribute of the input.'),
+    onChange: PropTypes.func.description(
+      `Function that will be called when the user types or pastes text.`,
+    ),
+    mask: PropTypes.arrayOf(
+      PropTypes.shape({
+        length: PropTypes.oneOfType([
+          PropTypes.number,
+          PropTypes.arrayOf(PropTypes.number),
+        ]),
+        fixed: PropTypes.string,
+        options: PropTypes.arrayOf(PropTypes.string),
+        regexp: PropTypes.shape({}), // RegExp
+      }),
+    ).description(
+      `Describes the structure of the mask. If a regexp is provided, it should
+      allow both the final full string element as well as partial strings
+      as the user types characters one by one.`,
```

Comment:
should we include a simple example here?

Reply:
I plan to put multiple rich examples in grommet-site.